src/eip7594/eip7594.c

```diff
@@ -709,11 +726,11 @@ static C_KZG_RET computed_weighted_sum_of_proofs(
     if (ret != C_KZG_OK) goto out;
 
     for (uint64_t i = 0; i < num_cells; i++) {
-        uint64_t pos = reverse_bits_limited(CELLS_PER_EXT_BLOB, cell_indices[i]);
-        fr_t coset_factor = s->roots_of_unity[pos];
-        // Compute h_k^n, with h_k and n as in the spec.
-        fr_pow(&coset_factor_pow, &coset_factor, FIELD_ELEMENTS_PER_CELL);
-        // Scale the power of r by h_k^n
+        /* Compute h_k^n, with h_k and n as in the spec */
+        uint64_t pos = CELL_INDICES_RBL[cell_indices[i]];
+        coset_factor_pow = s->roots_of_unity[pos * FIELD_ELEMENTS_PER_CELL];
```

Comment:
IMO this is not a straightforward change. I understand what we are trying to do, but not sure exactly why this should work. Is it possible to introduce a function similar to `coset_shift_for_cell()` from the spec (feel free to have our own naming) which does this logic with a bit of documentation?

Reply:
Hmm yes & no. If we did that, we'd have to manually take the power of it again. And it would only be used once.